service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/RequestBuilderService.java

```diff
@@ -1,21 +1,101 @@
 package uk.nhs.adaptors.gp2gp.common.service;
 
+import java.security.KeyStore;
+import java.util.ArrayList;
+import java.util.UUID;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+
 import io.netty.handler.ssl.SslContext;
 import io.netty.handler.ssl.SslContextBuilder;
 import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import lombok.RequiredArgsConstructor;
 import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.gpc.configuration.GpcConfiguration;
+import uk.nhs.adaptors.gp2gp.gpc.exception.GpConnectException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.web.reactive.function.client.ExchangeStrategies;
 
+import com.heroku.sdk.EnvKeyStore;
+
 @Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Slf4j
 public class RequestBuilderService {
     private static final int BYTE_COUNT = 16 * 1024 * 1024;
 
+    private final GpcConfiguration gpcConfiguration;
+
     @SneakyThrows
     public SslContext buildSSLContext() {
+        if (shouldBuildSslContext()) {
+            return buildSSLContextWithClientCertificates();
+        }
+        return SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE).build();
```

Comment:
We shouldn't build InsecureTrustManagerFactory by default.

Secure by default, not insecure by default! :D 

Why do we need InsecureTrustManagerFactory factory at all?